apps/web/src/contexts/theme-context.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react'
+
+const DEFAULT_THEME = 'light'
+
+/** Returns the theme set in localStorage, else the OS preferred theme */
+function getThemeFromBrowser(): string {
+  return (
+    localStorage.getItem('theme') ??
+    (window.matchMedia('(prefers-color-scheme: dark)').matches
+      ? 'dark'
+      : 'light')
+  )
+}
+
+/** Sets 'dark' class & css variables on <html>  */
+const applyTheme = (theme: string): void => {
+  const root = document.documentElement
+
+  if (theme === 'dark') {
+    root.classList.add('dark')
+  } else {
+    root.classList.remove('dark')
+  }
+}
+
+interface IThemeContext {
+  theme?: string
+  toggleTheme: () => void
+  setTheme: (theme?: string) => void
+}
+
+const ThemeContext = createContext<IThemeContext>({
+  theme: DEFAULT_THEME,
+  setTheme: () => {}, // eslint-disable-line @typescript-eslint/no-empty-function
+  toggleTheme: () => {}, // eslint-disable-line @typescript-eslint/no-empty-function
```

Comment:
Would it make sense to return null or add a comment inside to avoid the exception? Just an idea, this works too.